R/estimate_cox.R

```diff
@@ -0,0 +1,186 @@
+#'    The result is an object of class \code{coxph} which can be used in downstream functions and methods that rely on the \code{coxph} class.
+#'
+#' @param data The name of the dataset for the Cox model is based on the Analysis Data Model (ADaM) principles. The dataset is expected to have
+#'    one record per subject per analysis parameter. Rows in which the analysis variable (AVAL) or the sensor variable (CNSR) contain NA, are removed during analysis.
+#' @param equation Character vector, representing the formula to be estimated with the Cox model. When NULL, the baseline (null) model is fit.
+#'    Default is NULL.
+#' @param AVAL Analysis value for the Cox model. Default is "AVAL", as per CDISC ADaM guiding principles.
+#' @param CNSR Censor for the Cox model. Default is "CNSR", as per CDISC ADaM guiding principles.
+#' @param conf.int Confidence intervals used for the estimation. Conf.int argument is used from the \code{survival::summary.coxph}. The default is 0.95.
+#' @param ... additional arguments passed on to the ellipsis of the call \code{survival::coxph(data = data, formula = survival::Surv(AVAL, 1-CNSR) ~ equation), ...)}.
+#'    Use \code{?survival::coxph} and \code{?survival::summary.coxph} for more information.
+#'
+#' @inheritParams survival::coxph
+#' @inheritParams survival::summary.coxph
+#'
+#' @return an object of coxph, with additional components of the class of\code{survival::summary.coxph}. Use \code{?survival::coxph} and \code{?survival::summary.coxph} for more information.
+#'
+#' @references \url{https://github.com/therneau/survival}
+#'
+#' @export
+#'
+#' @examples
+#'
+#' ## Null model
+#' visR::estimate_cox(data = adtte)
+#'
+#' ## Univariate Cox model using the covariate: `TRTP`
+#' # Stratified Kaplan-Meier analysis by `TRTP`
+#' visR::estimate_cox(data = adtte, equation = "TRTP")
+#'
+#' ## Multivariate Cox model using the covariates: `TRTP` and `SEX`
+#' visR::estimate_cox(data = adtte, equation = "TRTP + SEX")
+#'
+#' ## Stratified Cox model (strata with three levels)
+#' visR::estimate_cox(data = adtte, equation = "AGE + strata(RACE)")
+#'
+#' ## Null model on subset of adtte
+#' visR::estimate_cox(data = adtte[adtte$SEX == "F", ])
+#'
+#' ## Cox model with cluster and different tie
+#' visR::estimate_cox(data = adtte, equation = "SEX + cluster(USUBJID)",
+#'   ties = "breslow", conf.int = 0.9)
+#'
+#' ## Example working with non CDISC data
+#' head(survival::veteran)
+#'
+#' # convert time and censoring data to ADaM variables
+#' # convert censoring status to CDISC principles
+#' veteran_adam <- survival::veteran %>%
+#'  dplyr::mutate(AVAL = time,
+#'                CNSR = dplyr::if_else(status == 1, 0, 1)
+#'  )
+#'
+#' visR::estimate_cox(data = veteran_adam, equation = "trt + diagtime")
+
+
+
+estimate_cox <- function(
+  data = NULL
+  ,equation = NULL
+  ,CNSR = "CNSR"
+  ,AVAL = "AVAL"
+  ,conf.int = 0.95
+  ,...
+){
```

Comment:
TBD - could we change equation to formula to be consistent with estimate_KM and general R terminology. I think we could be ok with not displaying the y side when CDISC, but also provides space to then have the complete formula for non CDISC data